esapi/api.bulk.go

```diff
@@ -143,6 +146,14 @@ func (r BulkRequest) Do(ctx context.Context, transport Transport) (*Response, er
 		req.Header[headerContentType] = headerContentTypeJSON
 	}
 
+	if len(r.Header) > 0 {
+		for k, vv := range r.Header {
```

Comment:
perhaps you could optimise the code path where `req.Header == nil`, and in that case set `req.Header = r.Header`?

Reply:
Right, that sounds like a good optimization!

Reply:
... although, on close inspection, I'm not sure if that would work, since the field is initialized in `newRequest()`:

https://github.com/elastic/go-elasticsearch/blob/dbbd052fde51991554bb0e183718c1bcda26b4bb/esapi/esapi.request.go#L36

```golang
h := make(http.Header)
fmt.Println(h == nil)
// => false
```

Reply:
`len(req.Header) == 0` then. Also, that means the nil checks in the generated code are unnecessary?

Reply:
Right, `len(req.Header)` is a good way to check it.

> the nil checks in the generated code are unnecessary

I think they're needed, otherwise the `r.Header.Add()` call would panic. (The `r` variable is `esapi.Request`, not `http.Request` coming from `newRequest` in the `Do()` method.)

https://github.com/elastic/go-elasticsearch/blob/c0687402022f213f62de6d78132f01a5e5242ddc/esapi/api.bulk.go#L299-L304

Reply:
Implemented in dc80821.